src/system/group.cr

```diff
@@ -0,0 +1,158 @@
+require "c/grp"
+require "c/sys/limits"
+
+struct System::Group
+  module Limits
+    {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+      GID_MAX = LibC::GID_MAX
+    {% elsif flag?(:linux) %}
+      GID_MAX = 0xffffffff_u32
+    {% else %}
+      GID_MAX = 0xffff_u32
+    {% end %}
+    NGROUPS_MAX = LibC::NGROUPS_MAX
+  end
+
+  alias GID = LibC::GidT
```

Comment:
Types shouldn't change based on platform. We should pick either 32 or 64 bit integers and bounds check them. This is different to C because in crystal we don't automatically convert between integer types so code *will* break once you  change types.

Reply:
Fixed in next push.